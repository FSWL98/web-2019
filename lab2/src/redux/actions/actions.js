export const ADD_NEW_FAVORITE_CITY = 'ADD_NEW_FAVORITE_CITY';
export const REMOVE_FAVORITE_CITY = 'REMOVE_FAVORITE_CITY';
export const ADD_ERROR_MODAL = 'ADD_ERROR_MODAL';
export const UPDATE_GEO_LOCATION = 'UPDATE_GEO_LOCATION';

export const addNewFavoriteCity = name => ({ type: ADD_NEW_FAVORITE_CITY, name: name });
export const removeFavoriteCity = name => ({ type: REMOVE_FAVORITE_CITY, name: name });
export const addErrorModal = errorMessage => ({ type: ADD_ERROR_MODAL, errorMessage: errorMessage });
export const updateGeoLocation = coordinates => ({ type: UPDATE_GEO_LOCATION, coordinates: coordinates });
