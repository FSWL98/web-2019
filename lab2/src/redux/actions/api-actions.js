export const API_GET_WEATHER_BY_CITY_NAME_REQUEST = "API_GET_WEATHER_BY_CITY_NAME_REQUEST";
export const API_GET_WEATHER_BY_CITY_NAME_RECEIVED = "API_GET_WEATHER_BY_CITY_NAME_RECEIVED";
export const API_GET_WEATHER_BY_CITY_COORDINATES_REQUEST = "API_GET_WEATHER_BY_CITY_COORDINATES_REQUEST";
export const API_GET_WEATHER_BY_CITY_COORDINATES_RECEIVED = "API_GET_WEATHER_BY_CITY_COORDINATES_RECEIVED";
export const API_ADD_COMPONENT_ID_TO_API = "API_ADD_COMPONENT_ID_TO_API";

export const addComponentIDTOApi = (componentID) => ({type: API_ADD_COMPONENT_ID_TO_API, componentID: componentID});

export const getWeatherByCityName = (cityname, fromID) => (
    { type: API_GET_WEATHER_BY_CITY_NAME_REQUEST, cityname: cityname, fromID: fromID });
export const getWeatherByCityCoordinates = (coordinates, fromID) => (
    {type: API_GET_WEATHER_BY_CITY_COORDINATES_REQUEST, coordinates: coordinates, fromID: fromID });
