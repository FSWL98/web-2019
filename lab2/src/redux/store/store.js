import { createStore } from 'redux';
import {rootReducer, initialState} from '../reducers/index';

const restoreDataFromLocalStorageByKey = (key) => {
    const storageData = localStorage.getItem(key);

    if (!storageData) {
        return null;
    }

    return JSON.parse(storageData);
};

const favoritesKeyStorage = "favorites";
const geoKeyStore = "geo";

const favorites = restoreDataFromLocalStorageByKey(favoritesKeyStorage);
const geo = restoreDataFromLocalStorageByKey(geoKeyStore);

let persistedState = initialState;

if (favorites) {
    persistedState.favourites = favorites;
}

if (geo) {
    persistedState.geo = geo;
}

export const store = createStore(
    rootReducer,
    persistedState,
    window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
);

store.subscribe(()=>{
    localStorage.setItem(favoritesKeyStorage, JSON.stringify(store.getState().favourites));
    localStorage.setItem(geoKeyStore, JSON.stringify(store.getState().geo));
});

export default store;
