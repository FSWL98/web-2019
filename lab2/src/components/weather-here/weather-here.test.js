import React from "react";
import WeatherHere from './weather-here';
import {act, create} from "react-test-renderer";
import {Provider} from "react-redux";
import store from "../../redux/store/store";
import {API_GET_WEATHER_BY_CITY_NAME_RECEIVED} from "../../redux/actions/api-actions";
import {updateGeoLocation} from "../../redux/actions/actions";

it("Default WeatherHere - not loaded", () => {
    let component = null;
    act(() => {
        component = create(
            <Provider store={store}>
                <WeatherHere />
            </Provider>
        );
    });

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

it("WeatherHere - loaded", () => {
    const real_ans = {
        status: 0,
        message: null,
        weather: {
            cityinfo: {
                name: "Msk",
                temperature: 10,
                icon: "01n"
            },
            measurements: [
                {
                    name: "Wind",
                    text: `10 m/s`,
                    color: "blue"
                },
                {
                    name: "Clouds",
                    text: `20 %`,
                    color: "red"
                },
                {
                    name: "Pressure",
                    text: `30 hpa`,
                    color: "yellow"
                },
                {
                    name: "Humidity",
                    text: `40 %`,
                    color: "green"
                },
                {
                    name: "Coordinates",
                    text: `[50, 60]`,
                    color: "pink"
                }
            ],
        }
    };

    store.dispatch(updateGeoLocation({lat: 10, lon: 20}));
    store.dispatch({ type: API_GET_WEATHER_BY_CITY_NAME_RECEIVED,
        state: real_ans, fromID: "main-weather"});

    let component = null;
    act(() => {
        component = create(
            <Provider store={store}>
                <WeatherHere  />
            </Provider>
        );
    });

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
