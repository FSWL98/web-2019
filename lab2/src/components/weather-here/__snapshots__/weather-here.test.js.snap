// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default WeatherHere - not loaded 1`] = `
<div
  className="mb-5 container"
>
  <div
    className="m-0 weather-here d-flex align-items-center justify-content-between row"
  >
    <h1>
      Weather Here
    </h1>
    <button
      className="bg-success border-0 btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Update Geolocation
    </button>
    <div
      className="w-25 invisible"
    >
      f
    </div>
  </div>
  <div
    className="weather-block--big d-flex align-items-center justify-content-center"
  >
    <div
      className="preloader"
    />
  </div>
</div>
`;

exports[`WeatherHere - loaded 1`] = `
<div
  className="mb-5 container"
>
  <div
    className="m-0 weather-here d-flex align-items-center justify-content-between row"
  >
    <h1>
      Weather Here
    </h1>
    <button
      className="bg-success border-0 btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Update Geolocation
    </button>
    <div
      className="w-25 invisible"
    >
      f
    </div>
  </div>
  <div
    className="p-0 d-flex container"
  >
    <div
      className="p-0  col-md-7"
    >
      <div
        className="container"
      >
        <div
          className="row"
        >
          <p
            className="city-temperature-geo-block__city-name"
          >
            Msk
          </p>
        </div>
        <div
          className="row"
        >
          <div
            className="col-md-4"
          >
            <img
              alt="icon"
              className="city-temperature-geo-block__icon"
              src="https://openweathermap.org/img/wn/01n@2x.png"
            />
          </div>
          <div
            className="col"
          >
            <span
              className="city-temperature-geo-block__temperature"
            >
              10
              Â°C
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="p-0 d-flex align-items-center col"
    >
      <div
        className="container"
      >
        <div
          className="measure-block-item measure-block-item--blue d-flex mt-md-1 justify-content-between row"
        >
          <div
            className="col"
          >
            Wind
          </div>
          <div
            className="d-flex justify-content-end font-weight-light col"
          >
            10 m/s
          </div>
        </div>
        <div
          className="measure-block-item measure-block-item--red d-flex mt-md-1 justify-content-between row"
        >
          <div
            className="col"
          >
            Clouds
          </div>
          <div
            className="d-flex justify-content-end font-weight-light col"
          >
            20 %
          </div>
        </div>
        <div
          className="measure-block-item measure-block-item--yellow d-flex mt-md-1 justify-content-between row"
        >
          <div
            className="col"
          >
            Pressure
          </div>
          <div
            className="d-flex justify-content-end font-weight-light col"
          >
            30 hpa
          </div>
        </div>
        <div
          className="measure-block-item measure-block-item--green d-flex mt-md-1 justify-content-between row"
        >
          <div
            className="col"
          >
            Humidity
          </div>
          <div
            className="d-flex justify-content-end font-weight-light col"
          >
            40 %
          </div>
        </div>
        <div
          className="measure-block-item measure-block-item--pink d-flex mt-md-1 justify-content-between row"
        >
          <div
            className="col"
          >
            Coordinates
          </div>
          <div
            className="d-flex justify-content-end font-weight-light col"
          >
            [50, 60]
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
